dump_queue_to_files.py

```diff
@@ -0,0 +1,72 @@
+import argparse
+import functools
+import shutil
+import uuid
+from datetime import datetime
+from pathlib import Path
+
+from google.cloud import storage
+
+from rabbit_context import RabbitContext
+
+
+def start_listening_to_rabbit_queue(queue, on_message_callback, timeout=30):
+    rabbit = RabbitContext(queue_name=queue)
+    connection = rabbit.open_connection()
+
+    connection.call_later(
+        delay=timeout,
+        callback=functools.partial(_timeout_callback, rabbit))
+    rabbit.channel.basic_consume(
+        queue=queue,
+        on_message_callback=on_message_callback)
+    rabbit.channel.start_consuming()
+
+
+def _timeout_callback(rabbit):
+    print('Message queue appears to be drained')
+    rabbit.close_connection()
+
+
+def dump_messages(queue_name, output_file_path):
+    directory_path = output_file_path.joinpath(f'{queue_name}_{datetime.utcnow().strftime("%Y-%M-%dT%H-%M-%S")}')
+    directory_path.mkdir()
+    start_listening_to_rabbit_queue(queue_name,
+                                    functools.partial(_rabbit_message_received_callback,
+                                                      output_file_path=directory_path))
+    return directory_path
+
+
+def _rabbit_message_received_callback(ch, method, _properties, body, output_file_path):
+    output_file = output_file_path.joinpath(f'{str(uuid.uuid4())}.json')
+    output_file.write_text(body.decode("utf-8"))
+    ch.basic_ack(delivery_tag=method.delivery_tag)
+
+
+def copy_file_to_gcs(file_path: Path):
+    client = storage.Client()
+    bucket = client.get_bucket(f'{client.project}-queue-dump-files')
```

Comment:
In the README it shows the format of the bucket name is `<PROJECT_ID>-<queue name>-queue-dump-files`. Does this need to match here too?

Reply:
fixed